apparmor/README.md

```diff
@@ -1,103 +1,487 @@
-security profile control -- for example, AppArmor can restrict file operations on specified paths.
+> **Difficulty**: Advanced
 
+> **Time**: Approximately 25 minutes
 
-By default, Docker runs containers with the `docker-default` AppArmor profile, which is described in the [documentation here](https://docs.docker.com/engine/security/apparmor/#understand-the-policies).  Here are some quick pointers for how to understand AppArmor profiles:
-  
-  - The include statements (such `#include <abstractions/base>`) behave just like their C-looking counterparts,
-  by expanding to additional AppArmor profile contents
+AppArmor (Application Armor) is a Linux Security Module (LSM). It protects the operating system by applying profiles to individual applications. In contrast to managing *capabilities* with `CAP_DROP` and syscalls with *seccomp*, AppArmor allows for much finer-grained control. For example, AppArmor can restrict file operations on specified paths.
 
-  - AppArmor is a deny-first system, specified by the `deny` clause.  Once a path or other resource is denied, it is impossible
-  to regain access to it with an `owner` statement in the same profile
+In this lab you will learn the basics of AppArmor and how to use it with Docker for improved security.
 
-  - For file operations, `r` corresponds to read, `w` to write, `k` to lock, `l` to link, and `x` to execute
+You will complete the following steps as part of this lab.
 
- This should get you started fairly well with AppArmor, but for more information you can consult the official [AppArmor documentation wiki](http://wiki.apparmor.net/index.php/Documentation) (under active development at this time).
+- [Step 1 - AppArmor primer](#primer)
+- [Step 2 - The `default-docker` AppArmor profile](#default-docker)
+- [Step 3 - Running a Container without an AppArmor profile](#no-profile)
+- [Step 4 - AppArmor and defense in depth](#depth)
+- [Step 5 - Custom AppArmor profile](#custom)
+- [Step 6 - Extra for experts](#extra)
 
-## Default AppArmor in Docker
+# Prerequisites
 
-1.  We can view the status of AppArmor on our Docker host by running `apparmor_status` (may require admin credentials) -- you should notice the `docker-default` profile is in enforce mode.  As described above, this is Docker's default AppArmor profile that is applied to containers on `docker run`.
+You will need all of the following to complete this lab:
 
-2.  To prove that this profile is applied by default, run an alpine container in another terminal `docker run --rm -it alpine sh` and then check `apparmor_status` again.  You should now see that a process (our alpine container!) has a profile defined and is also in enforce mode:
-    ```
-    1 processes have profiles defined.
-    1 processes are in enforce mode.
-       docker-default (28462)
-    ```
+- A Linux-based Docker Host with AppArmor enabled in the kernel
```

Comment:
Is it worth directing people to Debian based distros where AppArmor is more standard vs distros that prefer SELinux?